usr/src/uts/common/fs/zfs/metaslab.c

```diff
@@ -1039,6 +1039,13 @@ metaslab_group_allocatable(metaslab_group_t *mg, metaslab_group_t *rotor,
 		if (mg->mg_no_free_space)
 			return (B_FALSE);
 
+		/*
+		 * Relax allocation throttling for ditto blocks.  Due to
+		 * random imbalances in allocation it tends to push copies
+		 * to one vdev, that looks a bit better at the moment.
+		 */
+		qmax = qmax * (4 + d) / 4;
```

Comment:
I found this math a little hard to follow.  Maybe this would be more clear:
```
// increase qmax by 1/4 for the 2nd DVA, and by 1/2 for the 3rd dva
qmax += qmax * d / 4;
```
How did you decide on 1/4 (as opposed to 1/8, or 10, or unlimited)?

Reply:
Your version should do the same.  I don't mind.

I chose it mostly empirically.  My first version of the patch was just ignoring throttling for copies completely, and it worked fine, but then I thought it may be not good to remove all the limits, since then it would always allocate copies to following vdevs, even if they are slow, so I ended up like this, and my tests shown no more copies allocated to the same top-level vdev.  But If you prefer some other logic, or think that strict use of following vdevs is good, it would work for me too.

Reply:
I haven't tested that much, neither I would expect there easily reproducible difference.  It was only a thought that it could make code less depending on specific values I put here.